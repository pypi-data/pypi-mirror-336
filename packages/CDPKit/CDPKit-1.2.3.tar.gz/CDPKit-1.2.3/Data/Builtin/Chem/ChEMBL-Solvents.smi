WATER	[OH2]
DICHLOROMETHANE	ClCCl
TRICHLOROMETHANE	ClC(Cl)Cl
ETHYL ACETATE	CCOC(=O)C
METHANOL	CO
PROPAN-2-OL	CC(C)O
ACETONE	CC(=O)C
DMSO	CS(=O)C
ETHANOL	CCO
