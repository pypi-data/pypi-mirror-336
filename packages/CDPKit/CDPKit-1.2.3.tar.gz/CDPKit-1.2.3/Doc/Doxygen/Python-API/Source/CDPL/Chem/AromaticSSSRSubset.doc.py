#
# This file is part of the Chemical Data Processing Toolkit
#
# Copyright (C) Thomas Seidel <thomas.seidel@univie.ac.at>
#
# This program is free software; you can redistribute it and/or
# modify it under the terms of the GNU Lesser General Public
# License as published by the Free Software Foundation; either
# version 2 of the License, or (at your option) any later version.
#
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
# Lesser General Public License for more details.
#
# You should have received a copy of the GNU Lesser General Public License
# along with this program; see the file COPYING. If not, write to
# the Free Software Foundation, Inc., 59 Temple Place - Suite 330,
# Boston, MA 02111-1307, USA.
#

##
# \brief Implements the extraction of the aromatic rings in the SSSR of a molecular graph.
# 
class AromaticSSSRSubset(FragmentList):

    ##
    # \brief Constructs an empty <tt>AromaticSSSRSubset</tt> instance.
    # 
    def __init__() -> None: pass

    ##
    # \brief Initializes a copy of the \e %AromaticSSSRSubset instance \a arom_sssr.
    # \param arom_sssr The \e %AromaticSSSRSubset instance to copy.
    # 
    def __init__(arom_sssr: AromaticSSSRSubset) -> None: pass

    ##
    # \brief Construct a <tt>AromaticSSSRSubset</tt> instance that contains all aromatic rings of the SSSR of the molecular graph <em>molgraph</em>.
    # 
    # \param molgraph The molecular graph for which to extract the aromatic rings.
    # 
    def __init__(molgraph: MolecularGraph) -> None: pass

    ##
    # \brief Replaces the current state of \a self with a copy of the state of the \c %AromaticSSSRSubset instance \a arom_sssr.
    # \param arom_sssr The \c %AromaticSSSRSubset instance to copy.
    # \return \a self
    # 
    def assign(arom_sssr: AromaticSSSRSubset) -> AromaticSSSRSubset: pass

    ##
    # \brief Replaces the current set of rings by the aromatic rings in the SSSR of the molecular graph <em>molgraph</em>.
    # 
    # \param molgraph The molecular graph for which to extract the aromatic rings.
    # 
    def extract(molgraph: MolecularGraph) -> None: pass
