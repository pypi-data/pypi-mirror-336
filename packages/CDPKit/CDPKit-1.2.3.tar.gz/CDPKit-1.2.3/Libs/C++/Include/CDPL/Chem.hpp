/* 
 * Chem.hpp 
 *
 * This file is part of the Chemical Data Processing Toolkit
 *
 * Copyright (C) 2003 Thomas Seidel <thomas.seidel@univie.ac.at>
 *
 * This library is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 2 of the License, or (at your option) any later version.
 *
 * This library is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this library; see the file COPYING. If not, write to
 * the Free Software Foundation, Inc., 59 Temple Place - Suite 330,
 * Boston, MA 02111-1307, USA.
 */

/**
 * \file
 * \brief A convenience header including everything that is defined in namespace CDPL::Chem.
 */

#ifndef CDPL_CHEM_HPP
#define CDPL_CHEM_HPP

#include "CDPL/Chem/Entity3D.hpp"
#include "CDPL/Chem/Entity3DContainer.hpp"
#include "CDPL/Chem/AtomContainer.hpp"
#include "CDPL/Chem/BondContainer.hpp"
#include "CDPL/Chem/MolecularGraph.hpp"
#include "CDPL/Chem/Molecule.hpp"
#include "CDPL/Chem/Atom.hpp"
#include "CDPL/Chem/Bond.hpp"
#include "CDPL/Chem/Reaction.hpp"
#include "CDPL/Chem/BasicReaction.hpp"
#include "CDPL/Chem/BasicMolecule.hpp"
#include "CDPL/Chem/BasicAtom.hpp"
#include "CDPL/Chem/BasicBond.hpp"
#include "CDPL/Chem/Fragment.hpp"
#include "CDPL/Chem/ElectronSystem.hpp"
#include "CDPL/Chem/AtomDictionary.hpp"
#include "CDPL/Chem/FragmentList.hpp"
#include "CDPL/Chem/ElectronSystemList.hpp"
#include "CDPL/Chem/Entity3DMapping.hpp"
#include "CDPL/Chem/AtomMapping.hpp"
#include "CDPL/Chem/BondMapping.hpp"
#include "CDPL/Chem/AtomBondMapping.hpp"
#include "CDPL/Chem/StringDataBlock.hpp"
#include "CDPL/Chem/MatchConstraintList.hpp"
#include "CDPL/Chem/StereoDescriptor.hpp"
#include "CDPL/Chem/MultiConfMoleculeInputProcessor.hpp"
#include "CDPL/Chem/DefaultMultiConfMoleculeInputProcessor.hpp"
#include "CDPL/Chem/Atom3DCoordinatesFunction.hpp"
#include "CDPL/Chem/AtomPriorityFunction.hpp"
#include "CDPL/Chem/AtomPredicate.hpp"
#include "CDPL/Chem/BondPredicate.hpp"
#include "CDPL/Chem/AtomCompareFunction.hpp"
#include "CDPL/Chem/BondCompareFunction.hpp"
#include "CDPL/Chem/MatchExpression.hpp"
#include "CDPL/Chem/MatchExpressionList.hpp"
#include "CDPL/Chem/ANDMatchExpressionList.hpp"
#include "CDPL/Chem/ORMatchExpressionList.hpp"
#include "CDPL/Chem/NOTMatchExpression.hpp"
#include "CDPL/Chem/PropertyMatchExpression.hpp"
#include "CDPL/Chem/AtomTypeMatchExpression.hpp"
#include "CDPL/Chem/AtomEnvironmentMatchExpression.hpp"
#include "CDPL/Chem/AtomConfigurationMatchExpression.hpp"
#include "CDPL/Chem/AtomSSSRRingSizeMatchExpression.hpp"
#include "CDPL/Chem/BondConfigurationMatchExpression.hpp"
#include "CDPL/Chem/BondDirectionMatchExpression.hpp"
#include "CDPL/Chem/BondSubstituentDirectionMatchExpression.hpp"
#include "CDPL/Chem/BondReactionCenterStatusMatchExpression.hpp"
#include "CDPL/Chem/MolecularGraphComponentGroupingMatchExpression.hpp"
#include "CDPL/Chem/ReactionAtomMappingMatchExpression.hpp"
#include "CDPL/Chem/ReactionComponentGroupingMatchExpression.hpp"
#include "CDPL/Chem/Entity3DProperty.hpp"
#include "CDPL/Chem/AtomProperty.hpp"
#include "CDPL/Chem/BondProperty.hpp"
#include "CDPL/Chem/MolecularGraphProperty.hpp"
#include "CDPL/Chem/ReactionProperty.hpp"
#include "CDPL/Chem/AtomPropertyDefault.hpp"
#include "CDPL/Chem/BondPropertyDefault.hpp"
#include "CDPL/Chem/MolecularGraphPropertyDefault.hpp"
#include "CDPL/Chem/ReactionPropertyDefault.hpp"
#include "CDPL/Chem/AtomPropertyFlag.hpp"
#include "CDPL/Chem/BondPropertyFlag.hpp"
#include "CDPL/Chem/ControlParameter.hpp"
#include "CDPL/Chem/ControlParameterDefault.hpp"
#include "CDPL/Chem/ControlParameterFunctions.hpp"
#include "CDPL/Chem/AtomConfiguration.hpp"
#include "CDPL/Chem/BondConfiguration.hpp"
#include "CDPL/Chem/CIPDescriptor.hpp"
#include "CDPL/Chem/MDLParity.hpp"
#include "CDPL/Chem/BondStereoFlag.hpp"
#include "CDPL/Chem/AtomType.hpp"
#include "CDPL/Chem/SybylAtomType.hpp"
#include "CDPL/Chem/MOL2MoleculeType.hpp"
#include "CDPL/Chem/MOL2ChargeType.hpp"
#include "CDPL/Chem/HybridizationState.hpp"
#include "CDPL/Chem/ReactionCenterStatus.hpp"
#include "CDPL/Chem/ReactionRole.hpp"
#include "CDPL/Chem/BondDirection.hpp"
#include "CDPL/Chem/RadicalType.hpp"
#include "CDPL/Chem/INCHIReturnCode.hpp"
#include "CDPL/Chem/MDLDataFormatVersion.hpp"
#include "CDPL/Chem/DataFormat.hpp"
#include "CDPL/Chem/TautomerizationType.hpp"
#include "CDPL/Chem/AtomMatchConstraint.hpp"
#include "CDPL/Chem/BondMatchConstraint.hpp"
#include "CDPL/Chem/MolecularGraphMatchConstraint.hpp"
#include "CDPL/Chem/ReactionMatchConstraint.hpp"
#include "CDPL/Chem/INCHIMoleculeInputHandler.hpp"
#include "CDPL/Chem/INCHIMolecularGraphOutputHandler.hpp"
#include "CDPL/Chem/JMEMoleculeInputHandler.hpp"
#include "CDPL/Chem/JMEReactionInputHandler.hpp"
#include "CDPL/Chem/JMEMolecularGraphOutputHandler.hpp"
#include "CDPL/Chem/JMEReactionOutputHandler.hpp"
#include "CDPL/Chem/MOLMoleculeInputHandler.hpp"
#include "CDPL/Chem/MOLMolecularGraphOutputHandler.hpp"
#include "CDPL/Chem/SDFMoleculeInputHandler.hpp"
#include "CDPL/Chem/SDFMolecularGraphOutputHandler.hpp"
#include "CDPL/Chem/RXNReactionInputHandler.hpp"
#include "CDPL/Chem/RXNReactionOutputHandler.hpp"
#include "CDPL/Chem/RDFReactionInputHandler.hpp"
#include "CDPL/Chem/RDFReactionOutputHandler.hpp"
#include "CDPL/Chem/SMILESMoleculeInputHandler.hpp"
#include "CDPL/Chem/SMILESReactionInputHandler.hpp"
#include "CDPL/Chem/SMILESMolecularGraphOutputHandler.hpp"
#include "CDPL/Chem/SMILESReactionOutputHandler.hpp"
#include "CDPL/Chem/SMARTSMoleculeInputHandler.hpp"
#include "CDPL/Chem/SMARTSReactionInputHandler.hpp"
#include "CDPL/Chem/SMARTSMolecularGraphOutputHandler.hpp"
#include "CDPL/Chem/SMARTSReactionOutputHandler.hpp"
#include "CDPL/Chem/CDFMoleculeInputHandler.hpp"
#include "CDPL/Chem/CDFMolecularGraphOutputHandler.hpp"
#include "CDPL/Chem/CDFReactionInputHandler.hpp"
#include "CDPL/Chem/CDFReactionOutputHandler.hpp"
#include "CDPL/Chem/MOL2MoleculeInputHandler.hpp"
#include "CDPL/Chem/MOL2MolecularGraphOutputHandler.hpp"
#include "CDPL/Chem/XYZMoleculeInputHandler.hpp"
#include "CDPL/Chem/XYZMolecularGraphOutputHandler.hpp"
#include "CDPL/Chem/CMLMoleculeInputHandler.hpp"
#include "CDPL/Chem/CMLMolecularGraphOutputHandler.hpp"
#include "CDPL/Chem/SDFGZMoleculeInputHandler.hpp"
#include "CDPL/Chem/SDFGZMolecularGraphOutputHandler.hpp"
#include "CDPL/Chem/SDFBZ2MoleculeInputHandler.hpp"
#include "CDPL/Chem/SDFBZ2MolecularGraphOutputHandler.hpp"
#include "CDPL/Chem/SMILESGZMoleculeInputHandler.hpp"
#include "CDPL/Chem/SMILESGZReactionInputHandler.hpp"
#include "CDPL/Chem/SMILESGZMolecularGraphOutputHandler.hpp"
#include "CDPL/Chem/SMILESGZReactionOutputHandler.hpp"
#include "CDPL/Chem/SMILESBZ2MoleculeInputHandler.hpp"
#include "CDPL/Chem/SMILESBZ2ReactionInputHandler.hpp"
#include "CDPL/Chem/SMILESBZ2MolecularGraphOutputHandler.hpp"
#include "CDPL/Chem/SMILESBZ2ReactionOutputHandler.hpp"
#include "CDPL/Chem/RDFGZReactionInputHandler.hpp"
#include "CDPL/Chem/RDFGZReactionOutputHandler.hpp"
#include "CDPL/Chem/RDFBZ2ReactionInputHandler.hpp"
#include "CDPL/Chem/RDFBZ2ReactionOutputHandler.hpp"
#include "CDPL/Chem/CDFGZMoleculeInputHandler.hpp"
#include "CDPL/Chem/CDFGZMolecularGraphOutputHandler.hpp"
#include "CDPL/Chem/CDFBZ2MoleculeInputHandler.hpp"
#include "CDPL/Chem/CDFBZ2MolecularGraphOutputHandler.hpp"
#include "CDPL/Chem/CDFGZReactionInputHandler.hpp"
#include "CDPL/Chem/CDFGZReactionOutputHandler.hpp"
#include "CDPL/Chem/CDFBZ2ReactionInputHandler.hpp"
#include "CDPL/Chem/CDFBZ2ReactionOutputHandler.hpp"
#include "CDPL/Chem/MOL2GZMoleculeInputHandler.hpp"
#include "CDPL/Chem/MOL2GZMolecularGraphOutputHandler.hpp"
#include "CDPL/Chem/MOL2BZ2MoleculeInputHandler.hpp"
#include "CDPL/Chem/MOL2BZ2MolecularGraphOutputHandler.hpp"
#include "CDPL/Chem/XYZGZMoleculeInputHandler.hpp"
#include "CDPL/Chem/XYZGZMolecularGraphOutputHandler.hpp"
#include "CDPL/Chem/XYZBZ2MoleculeInputHandler.hpp"
#include "CDPL/Chem/XYZBZ2MolecularGraphOutputHandler.hpp"
#include "CDPL/Chem/CMLGZMoleculeInputHandler.hpp"
#include "CDPL/Chem/CMLGZMolecularGraphOutputHandler.hpp"
#include "CDPL/Chem/CMLBZ2MoleculeInputHandler.hpp"
#include "CDPL/Chem/CMLBZ2MolecularGraphOutputHandler.hpp"
#include "CDPL/Chem/SDFGZMoleculeReader.hpp"
#include "CDPL/Chem/SDFGZMolecularGraphWriter.hpp"
#include "CDPL/Chem/SDFBZ2MoleculeReader.hpp"
#include "CDPL/Chem/SDFBZ2MolecularGraphWriter.hpp"
#include "CDPL/Chem/RDFGZReactionReader.hpp"
#include "CDPL/Chem/RDFGZReactionWriter.hpp"
#include "CDPL/Chem/RDFBZ2ReactionReader.hpp"
#include "CDPL/Chem/RDFBZ2ReactionWriter.hpp"
#include "CDPL/Chem/SMILESGZMoleculeReader.hpp"
#include "CDPL/Chem/SMILESGZMolecularGraphWriter.hpp"
#include "CDPL/Chem/SMILESBZ2MoleculeReader.hpp"
#include "CDPL/Chem/SMILESBZ2MolecularGraphWriter.hpp"
#include "CDPL/Chem/SMILESGZReactionReader.hpp"
#include "CDPL/Chem/SMILESGZReactionWriter.hpp"
#include "CDPL/Chem/SMILESBZ2ReactionReader.hpp"
#include "CDPL/Chem/SMILESBZ2ReactionWriter.hpp"
#include "CDPL/Chem/CDFGZMoleculeReader.hpp"
#include "CDPL/Chem/CDFGZMolecularGraphWriter.hpp"
#include "CDPL/Chem/CDFBZ2MoleculeReader.hpp"
#include "CDPL/Chem/CDFBZ2MolecularGraphWriter.hpp"
#include "CDPL/Chem/CDFGZReactionReader.hpp"
#include "CDPL/Chem/CDFGZReactionWriter.hpp"
#include "CDPL/Chem/CDFBZ2ReactionReader.hpp"
#include "CDPL/Chem/CDFBZ2ReactionWriter.hpp"
#include "CDPL/Chem/MOL2GZMoleculeReader.hpp"
#include "CDPL/Chem/MOL2GZMolecularGraphWriter.hpp"
#include "CDPL/Chem/MOL2BZ2MoleculeReader.hpp"
#include "CDPL/Chem/MOL2BZ2MolecularGraphWriter.hpp"
#include "CDPL/Chem/XYZGZMoleculeReader.hpp"
#include "CDPL/Chem/XYZGZMolecularGraphWriter.hpp"
#include "CDPL/Chem/XYZBZ2MoleculeReader.hpp"
#include "CDPL/Chem/XYZBZ2MolecularGraphWriter.hpp"
#include "CDPL/Chem/CMLGZMoleculeReader.hpp"
#include "CDPL/Chem/CMLGZMolecularGraphWriter.hpp"
#include "CDPL/Chem/CMLBZ2MoleculeReader.hpp"
#include "CDPL/Chem/CMLBZ2MolecularGraphWriter.hpp"
#include "CDPL/Chem/JMEMoleculeReader.hpp"
#include "CDPL/Chem/JMEReactionReader.hpp"
#include "CDPL/Chem/JMEMolecularGraphWriter.hpp"
#include "CDPL/Chem/JMEReactionWriter.hpp"
#include "CDPL/Chem/SMILESMoleculeReader.hpp"
#include "CDPL/Chem/SMILESReactionReader.hpp"
#include "CDPL/Chem/SMILESMolecularGraphWriter.hpp"
#include "CDPL/Chem/SMILESReactionWriter.hpp"
#include "CDPL/Chem/SMARTSMoleculeReader.hpp"
#include "CDPL/Chem/SMARTSReactionReader.hpp"
#include "CDPL/Chem/SMARTSMolecularGraphWriter.hpp"
#include "CDPL/Chem/SMARTSReactionWriter.hpp"
#include "CDPL/Chem/MOLMoleculeReader.hpp"
#include "CDPL/Chem/MOLMolecularGraphWriter.hpp"
#include "CDPL/Chem/SDFMoleculeReader.hpp"
#include "CDPL/Chem/SDFMolecularGraphWriter.hpp"
#include "CDPL/Chem/RXNReactionReader.hpp"
#include "CDPL/Chem/RXNReactionWriter.hpp"
#include "CDPL/Chem/RDFReactionReader.hpp"
#include "CDPL/Chem/RDFReactionWriter.hpp"
#include "CDPL/Chem/INCHIMoleculeReader.hpp"
#include "CDPL/Chem/INCHIMolecularGraphWriter.hpp"
#include "CDPL/Chem/CDFMoleculeReader.hpp"
#include "CDPL/Chem/CDFMolecularGraphWriter.hpp"
#include "CDPL/Chem/CDFReactionReader.hpp"
#include "CDPL/Chem/CDFReactionWriter.hpp"
#include "CDPL/Chem/MOL2MoleculeReader.hpp"
#include "CDPL/Chem/MOL2MolecularGraphWriter.hpp"
#include "CDPL/Chem/XYZMoleculeReader.hpp"
#include "CDPL/Chem/XYZMolecularGraphWriter.hpp"
#include "CDPL/Chem/CMLMoleculeReader.hpp"
#include "CDPL/Chem/CMLMolecularGraphWriter.hpp"
#include "CDPL/Chem/MoleculeReader.hpp"
#include "CDPL/Chem/MolecularGraphWriter.hpp"
#include "CDPL/Chem/ReactionReader.hpp"
#include "CDPL/Chem/ReactionWriter.hpp"
#include "CDPL/Chem/CompleteRingSet.hpp"
#include "CDPL/Chem/AromaticRingSet.hpp"
#include "CDPL/Chem/AromaticSSSRSubset.hpp"
#include "CDPL/Chem/SmallestSetOfSmallestRings.hpp"
#include "CDPL/Chem/ExtendedSSSR.hpp"
#include "CDPL/Chem/AromaticSubstructure.hpp"
#include "CDPL/Chem/CyclicSubstructure.hpp"
#include "CDPL/Chem/ConnectedSubstructureSet.hpp"
#include "CDPL/Chem/ComponentSet.hpp"
#include "CDPL/Chem/PiElectronSystemList.hpp"
#include "CDPL/Chem/SubstructureSearch.hpp"
#include "CDPL/Chem/ReactionSubstructureSearch.hpp"
#include "CDPL/Chem/CommonConnectedSubstructureSearch.hpp"
#include "CDPL/Chem/MaxCommonAtomSubstructureSearch.hpp"
#include "CDPL/Chem/MaxCommonBondSubstructureSearch.hpp"
#include "CDPL/Chem/AutomorphismGroupSearch.hpp"
#include "CDPL/Chem/MultiSubstructureSearch.hpp"
#include "CDPL/Chem/Reactor.hpp"
#include "CDPL/Chem/MorganNumberingCalculator.hpp"
#include "CDPL/Chem/CIPPriorityCalculator.hpp"
#include "CDPL/Chem/CIPConfigurationLabeler.hpp"
#include "CDPL/Chem/CanonicalNumberingCalculator.hpp"
#include "CDPL/Chem/HashCodeCalculator.hpp"
#include "CDPL/Chem/Atom2DCoordinatesCalculator.hpp"
#include "CDPL/Chem/Hydrogen3DCoordinatesCalculator.hpp"
#include "CDPL/Chem/BondStereoFlagCalculator.hpp"
#include "CDPL/Chem/BondOrderCalculator.hpp"
#include "CDPL/Chem/KekuleStructureCalculator.hpp"
#include "CDPL/Chem/SymmetryClassCalculator.hpp"
#include "CDPL/Chem/SurfaceAtomExtractor.hpp"
#include "CDPL/Chem/PatternAtomTyper.hpp"
#include "CDPL/Chem/SubstructureHistogramCalculator.hpp"
#include "CDPL/Chem/TautomerScore.hpp"
#include "CDPL/Chem/TautomerGenerator.hpp"
#include "CDPL/Chem/DefaultTautomerGenerator.hpp"
#include "CDPL/Chem/TautomerizationRule.hpp"
#include "CDPL/Chem/PatternBasedTautomerizationRule.hpp"
#include "CDPL/Chem/KetoEnolTautomerization.hpp"
#include "CDPL/Chem/ImineEnamineTautomerization.hpp"
#include "CDPL/Chem/NitrosoOximeTautomerization.hpp"
#include "CDPL/Chem/AmideImidicAcidTautomerization.hpp"
#include "CDPL/Chem/LactamLactimTautomerization.hpp"
#include "CDPL/Chem/KeteneYnolTautomerization.hpp"
#include "CDPL/Chem/NitroAciTautomerization.hpp"
#include "CDPL/Chem/PhosphinicAcidTautomerization.hpp"
#include "CDPL/Chem/SulfenicAcidTautomerization.hpp"
#include "CDPL/Chem/GenericHydrogen13ShiftTautomerization.hpp"
#include "CDPL/Chem/GenericHydrogen15ShiftTautomerization.hpp"
#include "CDPL/Chem/ResonanceStructureGenerator.hpp"
#include "CDPL/Chem/StereoisomerGenerator.hpp"
#include "CDPL/Chem/ChEMBLStandardizer.hpp"
#include "CDPL/Chem/ProtonationStateStandardizer.hpp"
#include "CDPL/Chem/FragmentGenerator.hpp"
#include "CDPL/Chem/RECAPRuleID.hpp"
#include "CDPL/Chem/RECAPAtomLabel.hpp"
#include "CDPL/Chem/RECAPFragmentGenerator.hpp"
#include "CDPL/Chem/BRICSRuleID.hpp"
#include "CDPL/Chem/BRICSAtomLabel.hpp"
#include "CDPL/Chem/BRICSFragmentGenerator.hpp"
#include "CDPL/Chem/BemisMurckoAnalyzer.hpp"
#include "CDPL/Chem/TopologicalEntityAlignment.hpp"
#include "CDPL/Chem/SpatialEntityAlignment.hpp"
#include "CDPL/Chem/Atom3DCoordinatesFunctor.hpp"
#include "CDPL/Chem/AtomConformer3DCoordinatesFunctor.hpp"
#include "CDPL/Chem/AtomArray3DCoordinatesFunctor.hpp"
#include "CDPL/Chem/UtilityFunctions.hpp"
#include "CDPL/Chem/AtomContainerFunctions.hpp"
#include "CDPL/Chem/Entity3DContainerFunctions.hpp"
#include "CDPL/Chem/AtomFunctions.hpp"
#include "CDPL/Chem/BondFunctions.hpp"
#include "CDPL/Chem/MolecularGraphFunctions.hpp"
#include "CDPL/Chem/MoleculeFunctions.hpp"
#include "CDPL/Chem/FragmentFunctions.hpp"
#include "CDPL/Chem/ReactionFunctions.hpp"

#endif // CDPL_CHEM_HPP
