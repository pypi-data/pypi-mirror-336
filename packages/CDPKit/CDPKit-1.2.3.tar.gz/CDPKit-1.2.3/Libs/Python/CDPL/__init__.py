##
# __init__.py  
#
# This file is part of the Chemical Data Processing Toolkit
#
# Copyright (C) 2003 Thomas Seidel <thomas.seidel@univie.ac.at>
#
# This program is free software; you can redistribute it and/or
# modify it under the terms of the GNU Lesser General Public
# License as published by the Free Software Foundation; either
# version 2 of the License, or (at your option) any later version.
#
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
# Lesser General Public License for more details.
#
# You should have received a copy of the GNU Lesser General Public License
# along with this program; see the file COPYING. If not, write to
# the Free Software Foundation, Inc., 59 Temple Place - Suite 330,
# Boston, MA 02111-1307, USA.
##


from __future__ import absolute_import

import os
import sys
import platform

if sys.version_info[1] >= 8 and platform.system() == 'Windows':
    if 'site-packages' in __file__:
        os.add_dll_directory(os.path.dirname(__file__))
    else:
        os.add_dll_directory(os.path.join(os.path.dirname(os.path.dirname(os.path.dirname(__file__))), '@CDPKIT_EXECUTABLE_INSTALL_DIR@'))

from ._cdpl import *

__version__ = VERSION_STRING
