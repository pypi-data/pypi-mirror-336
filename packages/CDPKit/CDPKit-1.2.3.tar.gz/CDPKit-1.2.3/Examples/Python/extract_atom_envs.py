#!/bin/env python

##
# extract_atom_envs.py 
#
# This file is part of the Chemical Data Processing Toolkit
#
# Copyright (C) 2003 Thomas Seidel <thomas.seidel@univie.ac.at>
#
# Permission to use, copy, modify, and/or distribute this software for any purpose with
# or without fee is hereby granted.
# THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES WITH REGARD
# TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS. IN
# NO EVENT SHALL THE AUTHOR BE LIABLE FOR ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL
# DAMAGES OR ANY DAMAGES WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER
# IN AN ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF OR IN
# CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
##


import sys

import CDPL.Chem as Chem


# extracts the structural environments of the atoms in the specified molecular graph and outputs them as SMILES strings
def printEnvironments(molgraph: Chem.MolecularGraph) -> None:
    Chem.calcBasicProperties(molgraph, False)  # calculate basic molecular properties (if not yet done)
   
    frag = Chem.Fragment()                     # for storing extracted atom environments
    
    print('- Atom environments (radius = 3 bonds)')
    
    for atom in molgraph.atoms:
        Chem.getEnvironment(atom, molgraph, 3, frag)     # extract environment of atom reaching out up to three bonds
        Chem.perceiveComponents(frag, False)             # perceive molecular graph components (required for SMILES generation)

        smiles = Chem.generateSMILES(frag, False, False) # generate non-canonical SMILES string with explicit hydrogen atoms
        
        print('Atom #%s: %s' % (str(molgraph.getAtomIndex(atom)), smiles))

def main() -> None:
    if len(sys.argv) < 2:
        sys.exit('Usage: %s <mol. input file>' % sys.argv[0])

    # create reader for input molecules (format specified by file extension)
    reader = Chem.MoleculeReader(sys.argv[1]) 
    
    # create an instance of the default implementation of the Chem.Molecule interface
    mol = Chem.BasicMolecule()
    
    # read and process molecules one after the other until the end of input has been reached
    try:
        while reader.read(mol): 
            try:
                printEnvironments(mol)
            except Exception as e:
                sys.exit('Error: processing of molecule failed: ' + str(e))
                
    except Exception as e: # handle exception raised in case of severe read errors
        sys.exit('Error: reading molecule failed: ' + str(e))

    sys.exit(0)
        
if __name__ == '__main__':
    main()
