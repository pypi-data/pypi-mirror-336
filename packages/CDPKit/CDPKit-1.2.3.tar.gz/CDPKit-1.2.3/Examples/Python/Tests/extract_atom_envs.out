- Atom environments (radius = 3 bonds)
Atom #0: Fc(c(c)[H])c(c)[H]
Atom #1: c1(F)c(c(cc(c1[H])[H])[H])[H]
Atom #2: c1(c(F)c(cc(c1[H])C)[H])[H]
Atom #3: c1(c(c(F)cc(c1C(O)(c)C)[H])[H])[H]
Atom #4: c1(c(c(cc(c1[H])[H])[H])[H])C(OC)(c(c)c)C(C)([H])[H]
Atom #5: C1(c(c(c)[H])c(c)[H])(OC(c(c1c(c)[H])c)([H])[H])C(C(C)([H])[H])([H])[H]
Atom #6: O1C(c(c)c)(c(c(C1([H])[H])c)c)C(C)([H])[H]
Atom #7: C1(OC(c)(c(c1c(c)[H])c)C)([H])[H]
Atom #8: c12C(OC(c)(c1c(cc(c2[H])C)[H])C)([H])[H]
Atom #9: c12C(c(c)c)(OC(c1c(cc(c2[H])[H])[H])([H])[H])C(C)([H])[H]
Atom #10: c1(c(C(c)(O)C)c(C)cc(c1[H])C)[H]
Atom #11: c1(c(c(C)cc(c1C#N)[H])[H])[H]
Atom #12: c1(c(c(cc(C)c1[H])[H])[H])C#N
Atom #13: c1(c(C(O)([H])[H])c(C)cc(c1C#N)[H])[H]
Atom #14: [H]c(c(C)c)c(c)C
Atom #15: C(c(c(c)[H])c(c)[H])#N
Atom #16: N#Cc(c)c
Atom #17: [H]c(c(c)[H])c(c)C
Atom #18: [H]c(c(C)c)c(c)[H]
Atom #19: [H]C(OC)(c(c)c)[H]
Atom #20: [H]C(OC)(c(c)c)[H]
Atom #21: C(C(c(c)c)(OC)c(c)c)(C(C(N)([H])[H])([H])[H])([H])[H]
Atom #22: C(C(C(c)(O)c)([H])[H])(C(N(C)C)([H])[H])([H])[H]
Atom #23: C(C(C(C)([H])[H])([H])[H])(N(C([H])([H])[H])C([H])([H])[H])([H])[H]
Atom #24: N(C(C(C)([H])[H])([H])[H])(C([H])([H])[H])C([H])([H])[H]
Atom #25: C(N(C(C)([H])[H])C([H])([H])[H])([H])([H])[H]
Atom #26: [H]C(N(C)C)([H])[H]
Atom #27: [H]C(N(C)C)([H])[H]
Atom #28: [H]C(N(C)C)([H])[H]
Atom #29: C(N(C(C)([H])[H])C([H])([H])[H])([H])([H])[H]
Atom #30: [H]C(N(C)C)([H])[H]
Atom #31: [H]C(N(C)C)([H])[H]
Atom #32: [H]C(N(C)C)([H])[H]
Atom #33: [H]C(C(C)([H])[H])(N(C)C)[H]
Atom #34: [H]C(C(C)([H])[H])(N(C)C)[H]
Atom #35: [H]C(C(C)([H])[H])(C(N)([H])[H])[H]
Atom #36: [H]C(C(C)([H])[H])(C(N)([H])[H])[H]
Atom #37: [H]C(C(c)(O)c)(C(C)([H])[H])[H]
Atom #38: [H]C(C(c)(O)c)(C(C)([H])[H])[H]
Atom #39: c1(c(c(cc(F)c1[H])[H])C(O)(c)C)[H]
Atom #40: c1(c(F)c(cc(C)c1[H])[H])[H]
Atom #41: [H]c(c(F)c)c(c)[H]
Atom #42: [H]c(c(c)C)c(c)[H]
Atom #43: [H]c(c(c)[H])c(C)c
Atom #44: [H]c(c(F)c)c(c)[H]
